device_train.py

```diff
@@ -312,4 +315,8 @@ def eval_step(network, data):
             steps_per_sec = 1 / (time.time() - start)
             tokens_per_sec = tokens_per_step * steps_per_sec
 
-            wandb.log({'train/loss': loss, 'train/last_loss': last_loss, 'train/steps_per_sec': steps_per_sec, 'train/tokens_per_sec': tokens_per_sec}, step)
+            wandb.log({'train/loss': loss, 
+                'train/last_loss': last_loss, 
+                'train/steps_per_sec': steps_per_sec, 
+                'train/tokens_per_sec': tokens_per_sec,
+                'train/learning_rate': float(scheduler(step))}, step)
```

Comment:
I have a slight preference for using `network.state["opt_state"][-1].count().item()` in place of `step` here.

Why: the former is the actual step value optax uses when computing the lr.  It gets saved into checkpoints and loaded from them.

It _should_ always match the value of this script's `step` unless there's a bug.  But if there _is_ a bug, I'd prefer to report the real (and probably unintended) lr rather than the user-intended lr.

Reply:
thanks, yep better to be safe than sorry! fixed